Sources/Sentry/SentryTracer.m

```diff
@@ -672,7 +681,11 @@ - (SentryTransaction *)toTransaction
     NSMutableArray<id<SentrySpan>> *spans = [[NSMutableArray alloc] initWithCapacity:capacity];
 
     @synchronized(_children) {
-        [spans addObjectsFromArray:_children];
+        for (SentrySpan *child in _children) {
+            if (child.shouldIgnore == NO) {
+                [spans addObject:child];
+            }
```

Comment:
`h`: If I'm not mistaken, this approach won't work for nested UIViewControllers, as this would only remove the spans for `loadView` and `viewDidLoad` and not the other spans, such as TTID/TTFD.

Reply:
Sorry, TTID/TTFD was a bad example then. What about other spans, such as file IO, DB queries, manually added ones, or other future spans that we aren't yet aware of?

Reply:
Ok, I can see whether this could happen.
What do you think we should do to them?

Reply:
The validation idea mentioned in the top-level comment sounds promising 👍🏻

Reply:
We probably need to talk about this in person, maybe I didn't get it but your suggestion in the PR review comment does not solve the problem raised in here.

Also, I would not like to have a "scenario specific" logic inside the generic SentryTracer class, it feels more hacky than the one flag solution that indicates whether the span was validated by its creator or not.